packages/expo-ui/ios/Modifiers/ViewModifierRegistry.swift

```diff
@@ -826,6 +826,61 @@ internal struct ListRowBackground: ViewModifier, Record {
   }
 }
 
+internal enum ListRowSeparatorVisibility: String, Enumerable {
+  case automatic
+  case visible
+  case hidden
+
+  func toVisibility() -> Visibility {
+    switch self {
+    case .visible:
+      return .visible
+    case .hidden:
+      return .hidden
+    default:
+      return .automatic
+    }
+  }
+}
+
+internal enum VerticalEdgeOptions: String, Enumerable {
+  case all
+  case top
+  case bottom
+
+  func toVerticalEdges() -> VerticalEdge.Set {
+    switch self {
+    case .all:
+      return .all
+    case .top:
+      return .top
+    case .bottom:
+      return .bottom
+    }
+  }
+}
+
+internal struct ListRowSeparator: ViewModifier, Record {
+  @Field var visibility: ListRowSeparatorVisibility = .automatic
+  @Field var edges: VerticalEdgeOptions?
+
+  func body(content: Content) -> some View {
+    #if os(tvOS)
+      content
+    #else
+      if #available(iOS 15.0, macOS 12.0, tvOS 15.0, *) {
+        if let edges {
+          content.listRowSeparator(visibility.toVisibility(), edges: edges.toVerticalEdges())
+        } else {
+          content.listRowSeparator(visibility.toVisibility())
+        }
+      } else {
+        content
+      }
```

Comment:
```suggestion
      if let edges {
        content.listRowSeparator(visibility.toVisibility(), edges: edges.toVerticalEdges())
      } else {
        content.listRowSeparator(visibility.toVisibility())
      }
    }
```

we can remove the version check since min supported iOS is 15.0